src/components/Categories/EditCategory.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from "react"
+import { editCategory, getCategories, getCategory, postCategories } from "../../managers/categories"
+
+export const EditCategory = ({ categoryId, updateShowEditForm, categoryList, updateCategories }) => {
+    const [category, updateCategory] = useState({})
+
+    useEffect(() => {
+        getCategory(categoryId)
+            .then((categoryToEdit) => {
+                updateCategory(categoryToEdit)
+            })
+    }, [])
+
+
+    const handleSubmitCategory = (e) => {
+        e.preventDefault()
+
+        // Check if the category is already in database
+        const alreadyAdded = categoryList.some(existingCategory => existingCategory.label === category.label)
+
+        if (!alreadyAdded && category.label.length > 0) {
+            editCategory(category)
+                .then(updateCategories())
```

Comment:
If you change this to be an anonymous function `.then(() => updateCategories())` then it should refresh the page more reliably. I'm not entirely sure why it's working better, but from what I can gather, using the anonymous function ensures that the promise from the previous operation has been resolved before anything within the function can run.